x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/loadingservice/LocalModel.java

```diff
@@ -135,4 +134,39 @@ public void infer(Map<String, Object> fields, InferenceConfigUpdate update, Acti
         }
     }
 
+    public long acquire() {
+        return referenceCount.incrementAndGet();
```

Comment:
`referenceCount.incrementAndGet()` should always be > 1 correct? Do we want to prevent an acquire if the reference count has already reached zero?